collect_app/src/main/java/org/odk/collect/android/application/initialization/AppUpgrader.kt

```diff
@@ -0,0 +1,26 @@
+package org.odk.collect.android.application.initialization
+
+import org.odk.collect.shared.Settings
+
+class AppUpgrader(private val metaSettings: Settings, private val upgrades: List<Upgrade>) {
```

Comment:
I've potentially gone overboard on building out a small framework for calling upgrade code, but I found it really hard to slot the logic I needed in. My goal here was to separate the decision-making around whether we are upgrading or not from the different things that need to happen when we are.